### PR TITLE
Fix embedded tests

### PR DESCRIPTION
**Pull request by MattGill98**
Tests work for Payara 4 and 5, but to make Payara 4 work `-Dpayara4` is required.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -305,7 +311,38 @@
...
+            <properties>
+                <jersey.version>2.26</jersey.version>
+            </properties>
+        </profile>
+
+        <profile>
+            <id>payara-4</id>
+            <activation>
+                <property>
+                    <name>payara4</name>
```
Comment (Pandrex247) [inline pom.xml]: I don't like this.
I feel it's cleaner to just make 5 active by default, and then if you want to run 4 you specify so using the profile

Comment (MattGill98) [reply pom.xml]: If it's active by default, specifying -Ppayara-embedded overrides it and disables both profiles. So it's between `-Ppayara-embedded,payara4` and `-Ppayara-embedded -Dpayara4`. I personally preferred the latter as it allows you to ignore it when using Payara 5, hence me doing it like this

Comment (Pandrex247) [reply pom.xml]: I far prefer just being explicit and having to do `-Ppayara-embedded,payara4` and `-Ppayara-embedded,payara5` - it's how most of our other tests work

Comment (Pandrex247) [reply pom.xml]: Can you not just set the jersey property as the 5 version as "normal" (so no 5 profile), and only override it when you specify 4?

Comment (MattGill98) [reply pom.xml]: It now defaults to Payara 5 and will be reset to Payara 4 if the profile `payara4` is also specified.

**Pull request merged**